framework/db/CDbCommand.php

```diff
@@ -509,15 +509,19 @@ private function queryInternal($method,$mode,$params=array())
 			else
 				$this->_statement->execute($params);
 
-			if($method==='')
-				$result=new CDbDataReader($this);
-			else
-			{
+			if (in_array($method,['fetchColumn','fetch'])) {
 				$mode=(array)$mode;
 				call_user_func_array(array($this->_statement, 'setFetchMode'), $mode);
 				$result=$this->_statement->$method();
 				$this->_statement->closeCursor();
 			}
+			else {
+				$result = new CDbDataReader($this);
+				if (!empty($mode)) {
+					// $mode could be an array - so can't use just $result->setFetchMode($mode).
+					call_user_func_array([$result,'setFetchMode'],(array)$mode);
```

Comment:
Because in any case of using the result of fetching (either by `read()` and `readAll()` or as implementor of `Iterator` interface), it depends on such constructions as `$this->_statement->fetch()` or `$this->_statement->fetchAll()` which in turn depend of fetch mode - that is why we should take that option into account. Different values of that option mean different final result of the fetching - we can't just ignore it.